src/components/datepicker/DatePickerInput.tsx

```diff
@@ -0,0 +1,198 @@
+import React, { useEffect, useState } from 'react';
+import Box, { pickBoxProps } from '../box';
+import DatePicker from '.';
+import Icon from '../icon';
+import Input from '../input';
+import Popover from '../popover';
+import cx from 'classnames';
+import theme from './theme.css';
+import { formatDate } from './localeUtils';
+import { IconCalendarSmallOutline } from '@teamleader/ui-icons';
+import { DateTimeFormatOptions } from 'luxon';
+import { BoxProps } from '../box/Box';
+import { GenericComponent } from '../../@types/types';
+
+interface DatePickerInputProps extends Omit<BoxProps, 'size' | 'onChange'> {
+  /** A class name for the wrapper to give custom styles. */
+  className?: string;
+  /** Object with props for the DatePicker component. */
+  dayPickerProps?: DayPickerProps;
+  /** A footer component, rendered at the bottom of the date picker */
+  footer?: any;
+  /** A custom function to format a date. */
+  formatDate?: (selectedDate: Date, locale: string) => string;
+  /** Object with props for the Input component. */
+  inputProps?: TSFixMe; // TODO: change this to our input props
+  /** If true, component will be rendered in inverse mode. */
+  inverse?: boolean;
+  /** The language ISO locale code ('en-GB', 'nl-BE', 'fr-FR',...). */
+  locale?: string;
+  /** Callback function that is fired when the date has changed. */
+  onChange?: (selectedDate: Date) => void;
+  /** Callback function that is fired when the popover with the calendar gets closed (unfocused) */
+  onBlur?: () => void;
+  /** Object with props for the Popover component. */
+  popoverProps?: object;
```

Comment:
Probably putting this as TSFixMe would be better, so we don't forget after we transform Popover to TS.